kivy/tests/test_urlrequest.py

```diff
@@ -5,17 +5,50 @@
 import pytest
 import threading
 
+from datetime import datetime
 from time import sleep
 from base64 import b64encode
 import os
 
+if not os.environ.get('NONETWORK'):
+    from kivy.network.urlrequest import UrlRequest
+    from kivy.clock import Clock
```

Comment:
Can you use the `kivy_clock` fixture and pass it in, instead of importing it? 

Also, I don't like when we import kivy things globally in tests because then the kivy stuff is initialized before the tests start because kivy imports are so aggressive. It'd be better if you just do `from kivy.network.urlrequest import UrlRequest` in each test.

Reply:
I've updated the code as per your suggestion.
Out of curiosity, regarding kivy imports being aggressive. Dont't we have a deeper problem we want to fix one day? Or is it a feature that can't be fixed or that we don't want to fix?

Reply:
Yeah, we do have a very deep problem there that I hope will be fixed one day (the sooner the better). So this is just in preparation for that bright sunny morning :D